examples/heat_equation_spectral.cpp

```diff
@@ -158,12 +172,10 @@ int main(int argc, char** argv)
     // - the number of discrete time-points is equal to the number of
     //   steps + 1
     ddc::DiscreteDomain<DDimT> const time_domain
-            = ddc::init_discrete_space(
-                    DDimT::
-                            init(ddc::Coordinate<T>(start_time),
-                                 ddc::Coordinate<T>(end_time),
-                                 nb_time_steps + 1));
-    //! [time-domains]
+            = ddc::init_discrete_space<DDimT>(DDimT::init<DDimT>(
+                    ddc::Coordinate<T>(start_time),
+                    ddc::Coordinate<T>(end_time),
+                    nb_time_steps + 1));
```

Comment:
Is it necessary to include `DDimT` twice here? SFINAE doesn't work for:
```suggestion
            = ddc::init_discrete_space(DDimT::init<DDimT>(
                    ddc::Coordinate<T>(start_time),
                    ddc::Coordinate<T>(end_time),
                    nb_time_steps + 1));
```
?

Reply:
For now yes it is needed. We are thinking of improving that after documentation.